### PR TITLE
GOATMed: Toxic Rot

### PR DESCRIPTION
**Pull request by Risikio**
GOATMed

If someone has more than 75 Toxin Damage total running through their system, their day is going to get worse because as their internal organs begin to be damaged from the sheer amount of poisons running through your system.

Note:
Vaguely Tested. It works, however I don't know anything in regard to live server balance. 

I very helpfully labeled the variables to modify for balance reasons.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 1]
```diff
@@ -222,6 +222,15 @@
 			amount *= 0.5
 		..(amount)
 
+								//// TOXIN ORGAN ROT ////
+		if (ishuman(src))
+			if ((src.getToxLoss() > 75) && (amount>0)) // If toxloss is above a certain threshhold, more toxin damage will cause internal organ damage. For reference: 50 is DANGEROUS TOXIN LEVELS DETECTED
+				var/obj/item/organ/internal/targeted_organ
+				var/list/listed_organs  = list("brain",OP_EYES,"heart","lungs","stomach","liver","kidneys","appendix","psionic organ")
+				targeted_organ = src.random_organ_by_process(pick(listed_organs))
+				if (targeted_organ.nature !=MODIFICATION_SILICON) // If randomly chosen organ is prothestic, no damage.
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/human_damage.dm:231]: FPB buff confermed!

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 2]
```diff
@@ -222,6 +222,15 @@
 			amount *= 0.5
 		..(amount)
 
+								//// TOXIN ORGAN ROT ////
+		if (ishuman(src))
+			if ((src.getToxLoss() > 75) && (amount>0)) // If toxloss is above a certain threshhold, more toxin damage will cause internal organ damage. For reference: 50 is DANGEROUS TOXIN LEVELS DETECTED
+				var/obj/item/organ/internal/targeted_organ
+				var/list/listed_organs  = list("brain",OP_EYES,"heart","lungs","stomach","liver","kidneys","appendix","psionic organ")
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/human_damage.dm:229]: What happends if theirs missing one of these?

Comment (Risikio) [reply code/modules/mob/living/carbon/human/human_damage.dm:229]: It tumbles out of the proc and doesn't do anything. This was put in at the very end of the toxic damage proc.

**Comment by ThePainkiller** [comment]
As a long time medical player, can you please make it instead so that liver actually DOES ITS INTENDED JOB AND GETS DAMAGED FIRST, then the kidneys, and then MSOF?

As this stands, 75 toxloss will cause MSOF and it's going to be a nuissance more than actual good punishing balance

I know we're not baymed, but it would make more sense if liver/kidneys get damaged first (organ efficiency making it better at not getting too damaged by toxins), and then check for liver's organ integrity (HP) before starting to damage the rest of the organs in the body (The liver, who was in charge of holding back toxins damage, is completely dead => the rest of the system suffers from no filtering => poisoning worsens and causes widespread organ damage)

This would make:

- [x] More sense organ-wise
- [x] Make liver actually affect something for once as intended on any other med codebase
- [x] Medicines like Carthatoline actually useful for the heavy intoxication scenario they're specifically made for (since it also fixes liver damage)
- [x] Not push a Peridaxon meta to save on ATK's
- [x] Reviving someone who died of pure toxins damage less of a chore

**Comment by cdb-is-not-good** [comment]
That would be nice but there is one small problem with making it be liver based- and that's that Carth repairs liver damage and is already used by medical pretty much always

**Comment by Risikio** [comment]
Painkiller does have a point though, that with all of this there is going to be a push for a Peridaxon meta, and I honestly did consider having it target the kidneys first (since I don't think they actually do anything). 

However I also couldn't get through how to handle a synthetic liver since this toxin rot doesn't target synth organs. A single synth liver stops this entire system in its tracks... this system is sort of designed around the idea that there's just too much toxins for the liver to deal with.

Ultimately, when I'm done with what I'm trying to do involving organs, I'm going to **HIGHLY** suggest the implementation of the following...

Corpsmen being able to use an ATK to heal internal organs.

Tesselate Prayers to heal internal organs. Also Maybe a tesselate prayer to allow for painless "Psychic Surgery"? Maybe also pre-made 5u Peridoxin autoinjectors being able to be manufactured by the Church's bioprinters?

**Comment by Risikio** [comment]
Closing PR due to "Deluxe" PR going up.

**Pull request closed without merging**